sid/golang/assignment1/handlers/handlers.go

```diff
@@ -0,0 +1,106 @@
+package handlers
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/farkaskid/go_assignments/assignment1/helpers"
+)
+
+func Handle(cmd string, history *helpers.History) int {
+	// Main handler for commands, this will branch off to different handlers for different commands
+
+	commandFields := strings.Fields(cmd)
+
+	switch commandFields[0] {
+
+	case "ls":
+		return lsHandler(commandFields[1:])
+
+	case "cd":
+		return cdHander(commandFields[1:])
+
+	case "pwd":
+		return pwdHandler(commandFields[1:])
+
+	case "history":
+		return historyHandler(history)
+
+	default:
+		fmt.Println("Unknowm command:", cmd)
+		return 1
+	}
+}
+
+func lsHandler(inputs []string) int {
+	// Handles the ls command
+
+	currentDir, err := os.Open(".")
+
+	if err != nil {
+		helpers.PrintError("Failed to open current directory", err)
+
+		return 1
+	}
+
+	defer currentDir.Close()
+
+	files, err := currentDir.Readdirnames(0)
+
+	if err != nil {
+		helpers.PrintError("Failed to read current directory", err)
+
+		return 1
+	}
+
+	for _, file := range files {
+		fmt.Println(file)
+	}
+
+	return 0
+}
+
+func cdHander(input []string) int {
+	// Handler for the 'cd' command
+
```

Comment:
Handle cases like if no path is given or multiple paths given. Current code will throw panic.